Add HTTP module for downloading files

vlukash opened the pull request.

matthid commented:
Thanks a lot.
When we add a new `Fake.<>` namespace we need to add it here as well:

https://github.com/vlukash/FAKE/blob/1a8ba9ba009d94156a27982798ce9571ee113deb/build.fsx#L622-L631

This comment is more of a reminder for me before I merge this (but feel free to add it).

vlukash commented:
@matthid got it, thanks

Updated `build.fsx` and added functionality for parallel downloading
Ready for review, please share your thoughts on API/improvements

Current API is :

#### Single file download

```
downloadFile (localFilePath: string) (uri: string) : Result<string, string list> 
```
Usage example
```
Http.downloadFile "5.zip" @"http://ipv4.download.thinkbroadband.com/5MB.zip" |> ignore
```

#### Download in parallel:

```
downloadFiles (input: DownloadParameters list) : Result<string list, string list>
```
Usage example
```
let files: Http.DownloadParameters list = [
    {Path = "5.zip"; Uri = @"http://ipv4.download.thinkbroadband.com/5MB.zip"};
    {Path = "10.zip"; Uri = @"http://ipv4.download.thinkbroadband.com/10MB.zip"}]
let result = Http.downloadFiles files
match result with
    | Ok files ->
        printfn "%A" files
    | Error err ->
        printfn "%A" err
```

matthid commented:
My thoughts are:
 - It's good to have this ;)
 - I'd prefer throwing exceptions by default, because I think that is what other APIs are doing
 - We can have this as is but I think I add a `try` prefix to the function names
 - Namespace and module are fine (I like `Fake.Net.Http`)

The only other thing is if we want to have some docs for this, but we can improve this along the way.
Thanks a lot.
